src/k8s/pkg/snap/services.go

```diff
@@ -0,0 +1,107 @@
+package snap
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/canonical/k8s/pkg/utils"
+)
+
```

Comment:
what's the reason for the functions `GetServiceArgument` and `UpdateServiceArguments` to accept `Snap` as a first argument type rather than as a method on the `Snap`

`func (Snap *)GetServiceArgument(serviceName string ...) (...)`

Reply:
`Snap` is an interface, therefore it cannot implement methods.

Reply:
I guess Adam refers to the `snap` struct. 

If you do this, there would be no easy way to test this method. You could provide a mock implementation for other methods to use but you could not test that method in isolation. By injecting a mock `Snap` implementation this is possible.

Reply:
i think `Snap` as an interface was the reasoning i was looking for.  I don't have the go chops yet to understand the language semantics so this was just an effort to learn the language